app/models/presenters/tube_presenter.rb

```diff
@@ -60,5 +61,50 @@ def tubes_and_sources
         yield child_tubes if block_given? && child_tubes.present?
       end
     end
+
+    def qc_summary?
+      labware.receptacle&.qc_results&.to_a.present?
+    end
+
+    def qc_summary
+      labware.receptacle.qc_results.sort_by(&:key).each do |result|
+        yield result.key.titleize, result.unit_value.to_s
+      end
+      nil
```

Comment:
You shouldn't need to return nil here if you use
```
<% presenter.qc_summary do %>
```

Reply:
I am using `<% presenter.qc_summary do |key, value| %>`.  I copied the return of `nil` from https://github.com/sanger/limber/blob/develop/app/models/presenters/presenter.rb#L52

Should the `nil` be removed from there as well?  It was added by you on 5th Sep 2019.

Reply:
Turns out when you don't return `nil` the hash generated gets output directly to the UI

Reply:
Huh, okay, I thought that was an issue when you were using `<%=` rather than `<%`

I remember having this issue before, and adding the nil stuff as a work-around. I thought switching to `<%` was how I avoided needing to do it, but obviously there's something else at play that I'm forgetting.

Reply:
Keep the nil's for now, but I'll see if I can work out what's happening next time I'm in limber, as I'm pretty sure I've been successful in removing them elsewhere in the past.